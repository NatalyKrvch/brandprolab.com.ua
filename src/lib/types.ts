export type Size = 's' | 'm' | 'l';
export type Screen = 'mobile' | 'tablet' | 'desktop';
