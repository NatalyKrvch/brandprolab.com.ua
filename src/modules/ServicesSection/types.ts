import { ServicesSectionType } from '@/sanity/lib/types';

export interface ServicesSectionProps {
  servicesData: ServicesSectionType;
}

export type ServiceCardVariant = 'default' | 'gradient';
