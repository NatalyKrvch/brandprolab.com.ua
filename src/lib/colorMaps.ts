import {
  MainSocialCardColor,
  socialMediaColorClassMapType,
} from '@/components/Cards/SocialMediaCard/types';

export const socialMediaColorClassMap: socialMediaColorClassMapType = {
  [MainSocialCardColor.RED]: {
    bg: 'bg-red-default',
    text: 'text-red-default',
  },
  [MainSocialCardColor.BLUE]: {
    bg: 'bg-blue-default',
    text: 'text-blue-default',
  },
};
