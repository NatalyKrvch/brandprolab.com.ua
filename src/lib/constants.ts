export const ALL_DATA_QUERY = `
{
  "header": *[_type == "header"][0],
  "hero": *[_type == "hero"][0],
  "cases": *[_type == "cases"][0],
  "services": *[_type == "services"][0],
  "testimonials": *[_type == "testimonialsSection"][0],
  "subscribe": *[_type == "subscribeSection"][0],
  "video": *[_type == "videoSection"][0],
  "expertise": *[_type == "expertiseSection"][0],
  "diagnostic": *[_type == "diagnosticSection"][0],
  "footer": *[_type == "footerSection"][0]
}
`;

export const YOUTUBE_REGEX =
  /(?:youtu\.be\/|youtube\.com\/(?:.*[?&]v=|embed\/|v\/|shorts\/|live\/))([^&?/]+)/;

export enum IconClassType {
  SERVICES = 'services',
  EXPERTISE = 'expertise',
  SOCIAL_MEDIA = 'social-media',
  SOCIAL_MEDIA_MAIN = 'social-media-main',
}
export const REVALIDATE = 60;
export const ENDLESS_CAROUSEL_SPEED = 50;
export const SLIDER_RIGHT_SHIFT = 2;
export const SLIDER_LEFT_SHIFT = -1;
export const AMOUNT_OF_TESTIMONIALS_WORDS = 15;
export const AMOUNT_OF_CASES_WORDS_DESKTOP = 13;
export const AMOUNT_OF_CASES_WORDS_TABLET = 9;
export const AMOUNT_OF_CASES_WORDS_MOBILE = 13;
