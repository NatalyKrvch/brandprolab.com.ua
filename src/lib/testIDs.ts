export const INTERNAL_LINK_TEST_ID = 'internal-link-test-id';
export const EXTERNAL_LINK_TEST_ID = 'external-link-test-id';
export const FULL_WIDTH_CONTAINER_TEST_ID = 'full-width-container-test-id';
export const LIMITED_WIDTH_CONTAINER_TEST_ID =
  'limited-width-container-test-id';
export const LEFT_LIMITED_CONTAINER_TEST_ID = 'left-limited-container-test-id';
export const ICON_COMPONENT_TEST_ID = 'icon-component-test-id';
export const ICON_COMPONENT_IMAGE_TEST_ID = 'icon-component-image-test-id';
export const LABEL_COMPONENT_TEST_ID = 'label-component-test-id';
export const BUTTON_TEST_ID = 'button-test-id';
export const UNDERLINED_TEXT_TEST_ID = 'underlined-text-test-id';
export const USER_BADGE_TEST_ID = 'user-badge-test-id';
export const USER_BADGE_IMAGE_TEST_ID = 'user-badge-image-test-id';
export const USER_BADGE_NAME_TEST_ID = 'user-badge-name-test-id';
export const TESTIMONIAL_CARD_TEST_ID = 'testimonials-card-test-id';
export const TESTIMONIAL_TEXT_TEST_ID = 'testimonials-text-test-id';
export const LIST_TEST_ID = 'list-test-id';
export const LIST_ITEM_TEST_ID = 'list-item-test-id';
export const SIMPLE_CARD_TEST_ID = 'simple-card-test-id';
export const DIPLOMA_CARD_TEST_ID = 'diploma-card-test-id';
export const DIPLOMA_IMAGE_TEST_ID = 'diploma-image-test-id';
export const DIPLOMA_OVERLAY_TEST_ID = 'diploma-overlay-test-id';
export const PHOTO_CARD_IMAGE_TEST_ID = 'photo-card-image-test-id';
export const PHOTO_CARD_TEXT_TEST_ID = 'photo-card-text-test-id';
export const VIDEO_IFRAME_TEST_ID = 'video-iframe-test-id';
export const VIDEO_PLAY_BUTTON_TEST_ID = 'video-play-button-test-id';
export const VIDEO_TEST_ID = 'video-test-id';
export const VIDEO_THUMBNAIL_TEST_ID = 'video-thumbnail-test-id';
export const FLIP_CARD_BACK_TEST_ID = 'flip-card-back-test-id';
export const FLIP_CARD_FRONT_TEST_ID = 'flip-card-front-test-id';
export const FLIP_CARD_TEST_ID = 'flip-card-test-id';
export const SERVICE_CARD_FRONT_DESCRIPTION_TEST_ID =
  'service-card-front-description-test-id';
export const SERVICE_CARD_FRONT_HEADER_TEST_ID =
  'service-card-front-header-test-id';
export const SERVICE_CARD_BACK_HEADER_TEST_ID =
  'service-card-back-header-test-id';
export const SERVICE_CARD_BACK_LIST_ITEM_TEST_ID =
  'service-card-back-list-item-test-id';
export const SERVICE_CARD_BACK_LIST_TEST_ID = 'service-card-back-list-test-id';
export const SOCIAL_MEDIA_LABEL_TEST_ID = 'social-media-label-test-id';
export const SOCIAL_MEDIA_DESCRIPTION_TEST_ID =
  'social-media-description-test-id';
export const FOOTER_SECTION_TEST_ID = 'footer-section-test-id';
export const PAGINATION_DOT_TEST_ID = 'pagination-dot-test-id';
export const CAROUSEL_NAV_WRAPPER_TEST_ID = 'carousel-nav-wrapper-test-id';
export const CAROUSEL_NAV_NEXT_TEST_ID = 'carousel-nav-next-test-id';
export const CAROUSEL_NAV_PREV_TEST_ID = 'carousel-nav-prev-test-id';
export const ENDLESS_CAROUSEL_TEST_ID = 'endless-carousel-test-id';
export const ENDLESS_CAROUSEL_CONTENT_TEST_ID =
  'endless-carousel-content-test-id';
export const MAIN_SPRAY_BG_TEST_ID = 'main-spray-bg-test-id';
export const HERO_MAIN_CARD_ICON_TEST_ID = 'hero-main-card-icon-test-id';
export const HERO_TEXT_CONTENT_TEST_ID = 'hero-text-content-test-id';
export const HERO_TEXT_CONTENT_ICON_TEST_ID = 'hero-text-content-icon-test-id';
export const EXPERTISE_SECTION_TEST_ID = 'expertise-section-test-id';
export const SCROLL_UP_BUTTON_TEST_ID = 'scroll-up-button-test-id';
export const MODAL_WRAPPER_TEST_ID = 'modal-wrapper-test-id';
export const MODAL_CONTENT_TEST_ID = 'modal-content-test-id';
export const MODAL_CLOSE_BUTTON_TEST_ID = 'modal-close-button-test-id';
export const CASE_MODAL_TEXT_TEST_ID = 'case-modal-text-test-id';
export const TESTIMONIAL_MODAL_NAME_TEST_ID = 'testimonial-modal-name-test-id';
export const TESTIMONIAL_MODAL_POSITION_TEST_ID =
  'testimonial-modal-position-test-id';
export const TESTIMONIAL_MODAL_TEXT_TEST_ID = 'testimonial-modal-text-test-id';
export const TESTIMONIAL_MODAL_IMAGE_TEST_ID =
  'testimonial-modal-image-test-id';
