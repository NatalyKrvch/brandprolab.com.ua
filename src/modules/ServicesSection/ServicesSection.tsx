import { FlipCard, ServiceCardBack, ServiceCardFront } from '@/components';
import { normalizeImageURL, normalizeText } from '@/utils';

import { getCardVariantByIndex } from './helpers/getCardVariantByIndex';
import { serviceCardClassMap } from './styleMaps';
import type { ServicesSectionProps } from './types';

const ServicesSection = ({ servicesData }: ServicesSectionProps) => {
  const { title, description, cards, backgroundImage } = servicesData;

  return (
    <section className="flex flex-col gap-12">
      <div className="flex flex-col gap-4 px-1 text-left tablet:gap-4 tablet:px-2 desktop:gap-6 desktop:px-2">
        <h2 className="font-bold leading-9 -tracking-wide text-teal-dark tablet:text-5xl tablet:leading-48 desktop:text-6xl">
          {normalizeText(title)}
        </h2>

        <p className="max-w-530 text-base font-normal leading-5 tracking-wider text-gray-dark tablet:text-xl desktop:max-w-610 desktop:text-22 desktop:leading-8">
          {normalizeText(description)}
        </p>
      </div>

      <article className="grid grid-cols-1 items-stretch gap-4 mid_tablet:grid-cols-2 desktop:grid-cols-[1fr_1.302fr] desktop:gap-6">
        {cards.map((card, index) => {
          const variant = getCardVariantByIndex(index);
          const cardClass = serviceCardClassMap[variant];

          return (
            <FlipCard
              key={card._key}
              className={cardClass}
              front={
                <ServiceCardFront
                  iconURL={normalizeImageURL(card.icon)}
                  whiteIconURL={normalizeImageURL(card.whiteIcon ?? card.icon)}
                  callToActionText={card.flipText}
                  header={card.title}
                  label={card.label}
                  description={card.shortDescription}
                  backgroundImageUrl={normalizeImageURL(backgroundImage)}
                  variant={variant}
                />
              }
              back={
                <ServiceCardBack
                  header={card.detailedDescription.title}
                  list={card.detailedDescription.points}
                  backgroundImageUrl={normalizeImageURL(backgroundImage)}
                  variant={variant}
                />
              }
            />
          );
        })}
      </article>
    </section>
  );
};

export default ServicesSection;
