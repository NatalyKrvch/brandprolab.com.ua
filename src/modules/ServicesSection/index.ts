export { default as ServicesSection } from './ServicesSection';
export * from './types';
