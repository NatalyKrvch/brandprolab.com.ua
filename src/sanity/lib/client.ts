import { createClient } from 'next-sanity';

import { getSanityConfig } from '../env';

const { apiVersion, dataset, projectId, useCdn } = getSanityConfig();

export const client = createClient({
  projectId,
  dataset,
  apiVersion,
  useCdn,
});
