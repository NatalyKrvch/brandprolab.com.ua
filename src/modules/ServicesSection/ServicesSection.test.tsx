import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import { FLIP_CARD_TEST_ID } from '@/lib/testIDs';

import ServicesSection from './ServicesSection';

jest.mock('@/sanity/lib/client', () => ({
  client: {
    fetch: jest.fn(),
  },
}));

describe('ServicesSection component', () => {
  it('renders the component correctly', () => {
    const { asFragment } = render(
      <ServicesSection
        servicesData={{
          _id: 'mock-id',
          _type: 'services',
          title: 'Our Services',
          description: 'We offer various services to our clients',
          backgroundImage: {
            _type: 'image',
            asset: {
              _ref: 'image-ref',
              _type: 'reference',
            },
          },
          cards: [
            {
              _key: 'card-1',
              _type: 'card',
              icon: {
                _type: 'image',
                asset: {
                  _ref: 'icon-ref',
                  _type: 'reference',
                },
              },
              whiteIcon: {
                _type: 'image',
                asset: {
                  _ref: 'white-icon-ref',
                  _type: 'reference',
                },
              },
              title: 'Service 1',
              label: 'Label 1',
              shortDescription: 'Short description of service 1',
              flipText: 'Learn more',
              detailedDescription: {
                title: 'Details of service 1',
                points: ['Point 1', 'Point 2'],
              },
            },
          ],
        }}
      />,
    );

    const card = screen.getByTestId(FLIP_CARD_TEST_ID);
    expect(card).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
