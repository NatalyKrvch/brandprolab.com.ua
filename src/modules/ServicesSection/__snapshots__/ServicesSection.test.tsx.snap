// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesSection component renders the component correctly 1`] = `
<DocumentFragment>
  <section
    class="flex flex-col gap-12"
  >
    <div
      class="flex flex-col gap-4 px-1 text-left tablet:gap-4 tablet:px-2 desktop:gap-6 desktop:px-2"
    >
      <h2
        class="font-bold leading-9 -tracking-wide text-teal-dark tablet:text-5xl tablet:leading-48 desktop:text-6xl"
      >
        Наші послуги
      </h2>
      <p
        class="max-w-530 text-base font-normal leading-5 tracking-wider text-gray-dark tablet:text-xl desktop:max-w-610 desktop:text-22 desktop:leading-8"
      >
        Ми пропонуємо різні сервіси для клієнтів
      </p>
    </div>
    <article
      class="grid grid-cols-1 items-stretch gap-4 mid_tablet:grid-cols-2 desktop:grid-cols-[1fr_1.302fr] desktop:gap-6"
    >
      <div
        aria-label="Показати деталі"
        aria-pressed="false"
        class="group relative cursor-pointer overflow-hidden [perspective:10000px] h-full max-h-462 min-h-424 w-full min_mobile:max-h-462 min_mobile:min-h-462 tablet:max-h-424 tablet:min-h-400 desktop:h-500 desktop:max-h-500"
        data-testid="flip-card-test-id"
        role="button"
        tabindex="0"
      >
        <div
          class="relative h-full w-full transition-all duration-500 [transform-style:preserve-3d]"
        >
          <div
            class="absolute inset-0 flex h-full w-full [backface-visibility:hidden]"
            data-testid="flip-card-front-test-id"
          >
            <div
              class="relative w-full bg-cover bg-center bg-white border border-teal-dark rounded-24 tablet:rounded-32 desktop:rounded-40 flex flex-col gap-4 p-7 tablet:gap-0 mid_tablet:gap-0 desktop:gap-0 desktop:p-8"
              data-testid="simple-card-test-id"
            >
              <div
                aria-hidden="true"
                class="flex items-center justify-end"
              >
                <div
                  class="flex items-center justify-center block"
                  data-testid="icon-component-test-id"
                >
                  <div
                    class="bg-teal-darkOpacity relative flex items-center justify-center rounded-full w-68 h-68 tablet:w-80 tablet:h-80 desktop:w-108 desktop:h-108"
                  >
                    <img
                      alt=""
                      class="absolute w-16 h-16 tablet:w-72 tablet:h-72 desktop:w-96 desktop:h-96 -translate-x-24 tablet:-translate-x-30 desktop:-translate-x-36"
                      data-nimg="1"
                      data-testid="icon-component-image-test-id"
                      decoding="async"
                      height="0"
                      loading="lazy"
                      style="color: transparent;"
                      width="0"
                    />
                  </div>
                </div>
                <div
                  class="flex items-center justify-center hidden"
                  data-testid="icon-component-test-id"
                >
                  <div
                    class="bg-whiteOpacity relative flex items-center justify-center rounded-full w-68 h-68 tablet:w-80 tablet:h-80 desktop:w-108 desktop:h-108"
                  >
                    <img
                      alt=""
                      class="absolute w-16 h-16 tablet:w-72 tablet:h-72 desktop:w-96 desktop:h-96 -translate-x-24 tablet:-translate-x-30 desktop:-translate-x-36"
                      data-nimg="1"
                      data-testid="icon-component-image-test-id"
                      decoding="async"
                      height="0"
                      loading="lazy"
                      style="color: transparent;"
                      width="0"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex h-full flex-col justify-between"
              >
                <div
                  class="flex flex-col gap-4"
                >
                  <h3
                    class="max-w-300 px-1 text-26 font-bold leading-26 tablet:max-w-350 tablet:text-28 tablet:leading-7 desktop:max-w-350 desktop:text-32 desktop:leading-8 text-black"
                    data-testid="service-card-front-header-test-id"
                  >
                    Послуга 1
                  </h3>
                  <div
                    class="flex max-w-max items-center justify-center rounded-40 bg-teal-lightOpacity px-4 py-2 font-normal leading-14 tracking-normal text-blue-dark tablet:text-base tablet:leading-4 desktop:px-5 desktop:text-lg desktop:leading-18"
                    data-testid="label-component-test-id"
                  >
                    Етикетка 1
                  </div>
                  <p
                    class="px-1 pt-3 text-base font-normal leading-22 tablet:max-w-[500px] desktop:max-w-[500px] desktop:text-lg text-black"
                    data-testid="service-card-front-description-test-id"
                  >
                    Короткий опис послуги 1
                  </p>
                </div>
                <button
                  aria-label="Переглянути деталі про Послуга 1"
                  class="px-1 text-left desktop:px-2"
                >
                  <span
                    class="relative inline-block cursor-pointer text-base leading-5 text-teal-dark underline underline-offset-4 font-bold desktop:text-lg"
                    data-testid="underlined-text-test-id"
                  >
                    Дізнатись більше
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="absolute inset-0 flex h-full w-full [backface-visibility:hidden] [transform:rotateX(180deg)]"
            data-testid="flip-card-back-test-id"
          >
            <div
              class="relative w-full bg-cover bg-center rounded-24 tablet:rounded-32 desktop:rounded-40 flex h-full flex-col gap-4 bg-blue-dark p-6 tablet:p-8 mid_tablet:border-none desktop:gap-6 desktop:border-none desktop:p-10"
              data-testid="simple-card-test-id"
            >
              <h3
                class="mid_tablet:max-w-600 desktop:max-w-600 text-lg font-bold leading-5 text-teal-medium min_mobile:text-base tablet:text-2xl tablet:leading-6 mid_tablet:text-xl desktop:text-2xl desktop:leading-6"
                data-testid="service-card-back-header-test-id"
              >
                Деталі послуги 1
              </h3>
              <ul
                class="mid_tablet:max-w-540 desktop:max-w-544 list-disc space-y-1.5 pl-5 text-sm font-extralight leading-18 tracking-normal text-white min_mobile:text-xs tablet:text-base tablet:leading-6 mid_tablet:text-sm mid_tablet:leading-4 desktop:text-lg desktop:leading-5"
                data-testid="service-card-back-list-test-id"
              >
                <li
                  data-testid="service-card-back-list-item-test-id"
                >
                  Пункт 1
                </li>
                <li
                  data-testid="service-card-back-list-item-test-id"
                >
                  Пункт 2
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
